apps/ejabberd/src/mongoose_admin.erl

```diff
@@ -0,0 +1,264 @@
+            {function, kick_this_session},
+            {action, delete},
+            {args, [{jid, binary}]},
+            {result, {msg, binary}}
+        ],
+        [
+            {name, sendmessage},
+            {category, message},
+            {desc, "Send chat message from to"},
+            {module, ?MODULE},
+            {function, send_message},
+            {action, create},
+            {security_policy, [user]},
+            {args, [{caller, binary}, {to, binary}, {msg, binary}]},
+            {result, ok}
+        ],
+        [
+            {name, getmessages},
+            {category, message},
+            {desc, "Get recent messages"},
+            {module, ?MODULE},
+            {function, get_recent_messages},
+            {action, read},
+            {security_policy, [user]},
+            {args, [{caller, binary}, {other, binary}, {limit, integer}]},
+            {result, []}
+        ],
+        [
+            {name, changepassword},
+            {category, user},
+            {desc, "Change user password"},
+            {module, ?MODULE},
+            {function, change_user_password},
+            {action, update},
+            {security_policy, [user]},
+            {identifiers, [caller]},
+            {args, [{caller, binary}, {newpass, binary}]},
+            {result, ok}
+        ]
+    ].
+
+%%get_user_resources(Jid) ->
+%%    J = jid:from_binary(Jid),
+%%    ejabberd_sm:get_user_resources(J#jid.user, J#jid.server).
+
+
+kick_this_session(Jid) ->
+    J = jid:from_binary(Jid),
+    ejabberd_sm:route(
+        jid:make(<<"">>, <<"">>, <<"">>),
+        J,
+        {broadcast, {exit, <<"kicked">>}}),
+    <<"kicked">>.
+
+mk(E) ->
+    Ss = element(1, E),
+    {U, S, R} = Ss,
+    <<U/binary, "@", S/binary, "/", R/binary>>.
+
+listsessions(Host) ->
+    Lst = ejabberd_sm:get_vh_session_list(Host),
+    lists:map(fun mk/1, Lst).
+
+
+registered_users(Host) ->
+    Users = ejabberd_auth:get_vh_registered_users(Host),
+    SUsers = lists:sort(Users),
+    lists:map(fun({U, S}) -> <<U/binary, "@", S/binary>> end, SUsers).
+
+
+register(User, Host, Password) ->
+    case ejabberd_auth:try_register(User, Host, Password) of
+        ok ->
+            list_to_binary(io_lib:format("User ~s@~s successfully registered", [User, Host]));
+        {error, exists} ->
+            String = io_lib:format("User ~s@~s already registered at node ~p",
+                [User, Host, node()]),
+            throw({exists, String});
+        {error, Reason} ->
+            String = io_lib:format("Can't register user ~s@~s at node ~p: ~p",
+                [User, Host, node(), Reason]),
+            throw({error, String})
+    end.
+
+unregister(Jid) ->
+    J = jid:from_binary(Jid),
+    ejabberd_auth:remove_user(J#jid.luser, J#jid.lserver),
+    <<"">>.
+
+
+send_message(From, To, Body) ->
+    Packet = build_packet(message_chat, Body),
+    F = jid:from_binary(From),
+    T = jid:from_binary(To),
+    ejabberd_hooks:run(user_send_packet,
```

Comment:
I'd rather use new hook for that "rest_user_send_packet" in order to have better visibility from where the message was send.